drivers/bluetooth/hci/userchan.c

```diff
@@ -257,14 +291,18 @@ SYS_INIT(bt_uc_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
 
 static void cmd_bt_dev_found(char *argv, int offset)
 {
-	if (strncmp(&argv[offset], "hci", 3) || strlen(&argv[offset]) < 4) {
-		posix_print_error_and_exit("Error: Invalid Bluetooth device "
-					   "name '%s' (should be e.g. hci0)\n",
+	arg_found = true;
+	if (strncmp(&argv[offset], "hci", 3) == 0 && strlen(&argv[offset]) >= 4) {
+		/* It's a hciX string */
+		hci_socket = true;
+		bt_dev_index = strtol(&argv[offset + 3], NULL, 10);
+	} else if (sscanf(&argv[offset], "%15[^:]:%d", ip_addr, &port) == 2) {
```

Comment:
I wonder if this should have been 
```suggestion
	} else if (sscanf(&argv[offset], "%15[^:]:%d", ip_addr, &port) != 1) {
```

Reply:
Well, I am scanning for two arguments. The IP address and the port number. As I understand `sscanf` returns the number of matched arguments if succesful. In this case two.  By checking for minus one I am not sure if both arguments would be correct. Would it?

Reply:
I disagree. What we want to test here (given that the result is `hci_socket = false;` is whether the number of arguments are 1, right?

It is only a `hci_socket` the number of arguments is 1.

Reply:
> What we want to test here 

Is that it got what looks like a valid ip + port configuration (x.x.x.x:port)

Reply:
I'll concede :)